### PR TITLE
[Merged by Bors] - chore: replace `λ` by `fun`

### PR DESCRIPTION
**Pull request by grunweg**
Per the style guidelines, `λ` is disallowed in mathlib.
This is close to exhaustive; I left some tactic code alone when it seemed to me that tactic could be upstreamed soon.

Notes
- In lines I was modifying anyway, I also converted `=>` to `↦`.
- Also contains some mild in-passing indentation fixes in `Mathlib/Order/SupClosed`.
- Some doc comments still contained Lean 3 syntax `λ x, `, which I also replaced.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Drive-by question: is the Lean 3 syntax used in comments in `Data/FunLike/{Embedding,Equiv}` and `Data/Polynomial/Laurent` intended, or did somebody simply forget to fix this during the port?

File: Mathlib/Algebra/Field/ULift.lean [hunk 1]
```diff
@@ -21,7 +21,7 @@ variable {α : Type u} {x y : ULift.{v} α}
 
 namespace ULift
 
-instance [RatCast α] : RatCast (ULift α) := ⟨λ a ↦ up a⟩
+instance [RatCast α] : RatCast (ULift α) := ⟨fun a ↦ up a⟩
```
Comment (mcdoll) [inline Mathlib/Algebra/Field/ULift.lean:24]:
```suggestion
instance [RatCast α] : RatCast (ULift α) := ⟨(up ·)⟩
```

File: Mathlib/Algebra/Group/ULift.lean [hunk 2]
```diff
@@ -143,9 +143,9 @@ instance commMonoid [CommMonoid α] : CommMonoid (ULift α) :=
 #align ulift.comm_monoid ULift.commMonoid
 #align ulift.add_comm_monoid ULift.addCommMonoid
 
-instance natCast [NatCast α] : NatCast (ULift α) := ⟨λ a ↦ up a⟩
+instance natCast [NatCast α] : NatCast (ULift α) := ⟨fun a ↦ up a⟩
```
Comment (mcdoll) [inline Mathlib/Algebra/Group/ULift.lean:146]:
```suggestion
instance natCast [NatCast α] : NatCast (ULift α) := ⟨(up ·)⟩
```

File: Mathlib/Algebra/Group/ULift.lean [hunk 3]
```diff
@@ -143,9 +143,9 @@ instance commMonoid [CommMonoid α] : CommMonoid (ULift α) :=
...
 #align ulift.has_nat_cast ULift.natCast
-instance intCast [IntCast α] : IntCast (ULift α) := ⟨λ a ↦ up a⟩
+instance intCast [IntCast α] : IntCast (ULift α) := ⟨fun a ↦ up a⟩
```
Comment (mcdoll) [inline Mathlib/Algebra/Group/ULift.lean:148]:
```suggestion
instance intCast [IntCast α] : IntCast (ULift α) := ⟨(up ·)⟩
```
and on line 185 use `(⟨·⟩)` (note that the parentheses are outside the anonymous constructor)

File: Mathlib/Analysis/SpecificLimits/Basic.lean [hunk 4]
```diff
@@ -693,7 +693,7 @@ theorem tendsto_nat_ceil_div_atTop : Tendsto (fun x ↦ (⌈x⌉₊ : R) / x) at
   simpa using tendsto_nat_ceil_mul_div_atTop (zero_le_one' R)
 #align tendsto_nat_ceil_div_at_top tendsto_nat_ceil_div_atTop
 
-lemma Nat.tendsto_div_const_atTop {n : ℕ} (hn : n ≠ 0) : Tendsto (λ x ↦ x / n) atTop atTop := by
+lemma Nat.tendsto_div_const_atTop {n : ℕ} (hn : n ≠ 0) : Tendsto (fun x ↦ x / n) atTop atTop := by
```
Comment (mcdoll) [inline Mathlib/Analysis/SpecificLimits/Basic.lean:696]:
```suggestion
lemma Nat.tendsto_div_const_atTop {n : ℕ} (hn : n ≠ 0) : Tendsto (· / n) atTop atTop := by
```

File: Mathlib/Combinatorics/Quiver/Cast.lean [hunk 5]
```diff
@@ -32,7 +32,7 @@ namespace Quiver
 
 /-- Change the endpoints of an arrow using equalities. -/
 def Hom.cast {u v u' v' : U} (hu : u = u') (hv : v = v') (e : u ⟶ v) : u' ⟶ v' :=
-  Eq.ndrec (motive := λ x => x ⟶ v') (Eq.ndrec e hv) hu
+  Eq.ndrec (motive := fun x ↦ x ⟶ v') (Eq.ndrec e hv) hu
```
Comment (mcdoll) [inline Mathlib/Combinatorics/Quiver/Cast.lean:35]:
```suggestion
  Eq.ndrec (motive := (· ⟶ v')) (Eq.ndrec e hv) hu
```
untested

File: Mathlib/Combinatorics/Quiver/Cast.lean [hunk 6]
```diff
@@ -81,7 +81,7 @@ open Path
 
 /-- Change the endpoints of a path using equalities. -/
 def Path.cast {u v u' v' : U} (hu : u = u') (hv : v = v') (p : Path u v) : Path u' v' :=
-  Eq.ndrec (motive := λ x => Path x v') (Eq.ndrec p hv) hu
+  Eq.ndrec (motive := fun x ↦ Path x v') (Eq.ndrec p hv) hu
```
Comment (mcdoll) [inline Mathlib/Combinatorics/Quiver/Cast.lean:84]:
```suggestion
  Eq.ndrec (motive := (Path · v')) (Eq.ndrec p hv) hu
```

File: test/MfldSetTac.lean [hunk 7]
```diff
@@ -24,7 +24,7 @@ section stub_lemmas
 structure PartialHomeomorph (α : Type u) (β : Type u) extends PartialEquiv α β
 
 noncomputable
-instance PartialHomeomorph.has_coe_to_fun : CoeFun (PartialHomeomorph α β) (λ _ => α → β) := test_sorry
+instance PartialHomeomorph.has_coe_to_fun : CoeFun (PartialHomeomorph α β) (fun _ ↦  α → β) := test_sorry
```
Comment (mcdoll) [inline test/MfldSetTac.lean:27]: Here, below and in `test/toAdditive` you have a whitespace too much (maybe somewhere else as well?)

**Review by mcdoll — COMMENTED** [review]
I am pretty sure that we don't want any Lean 3 syntax anymore in the comments.
I am less sure whether we want to ban `λ` from meta-code.

File: test/MfldSetTac.lean (hunk 7 above)
Comment (grunweg) [reply test/MfldSetTac.lean]: Thanks; fixed (here and in all of mathlib). This means this PR has some drive-by space changes.

**Comment by grunweg** [comment]
I'll be happy to drop the changes to the `Lean/Meta` or `Tactic` directory, if I hear this is desired.
The style guide says "The lambda notation λ x ↦ x * x, while syntactically valid, is disallowed in mathlib in favor of the fun keyword.", which to me seems to include any mathlib tactics. (If this is not desired, this should be updated: happy to make a PR.)

**Comment by dupuisf** [comment]
Sorry for being cryptic :-) I meant that I agree that the Lean 3 syntax in the comments is almost certainly an artifact of the port and should have been converted to Lean 4 syntax.

**Comment by grunweg** [comment]
Thanks for elaborating. I just looked into this; five PRs later, this has been fixed. #11405 and #11409 are particular off-spring.

**Pull request closed without merging**